packages/core/src/scanning.ts

```diff
@@ -106,6 +96,52 @@ export const scanOutputs = (
         );
         n += counterIncrement;
     } while (counterIncrement > 0 && outputs.length > 0);
+}
+
+export const scanOutputs = (
+    scanPrivateKey: Buffer,
+    spendPublicKey: Buffer,
+    sumOfInputPublicKeys: Buffer,
+    inputHash: Buffer,
+    outputs: Buffer[],
+    labels?: LabelMap,
+): Map<string, Buffer> => {
+    const ecdhSecret = secp256k1.publicKeyTweakMul(
+        sumOfInputPublicKeys,
+        secp256k1.privateKeyTweakMul(scanPrivateKey, inputHash),
+        true,
+    );
+
+    // output to tweak data map
+    const matches = new Map<string, Buffer>();
+
+    performScanLoop(ecdhSecret, spendPublicKey, outputs, matches, labels);
+
+    return matches;
+};
```

Comment:
Once you start returning matches, we can just do
```suggestion
    return performScanLoop(ecdhSecret, spendPublicKey, outputs, labels);
```